diff -up ki18n-5.57.0/src/klocalizedstring.cpp.0000~ ki18n-5.57.0/src/klocalizedstring.cpp
--- ki18n-5.57.0/src/klocalizedstring.cpp.0000~	2019-04-10 22:36:16.053806583 +0200
+++ ki18n-5.57.0/src/klocalizedstring.cpp	2019-04-10 22:42:23.451360758 +0200
@@ -475,6 +475,13 @@ void KLocalizedStringPrivate::translateR
         } else {
             testMsgstr = catalog.translate(msgid);
         }
+	if (testMsgstr.isEmpty()) {
+            // Try translations from extra openmandriva-kde-translation catalog if none was found
+            const KCatalog &xcatalog = getCatalog("openmandriva-kde-translation", testLanguage);
+            if (!msgctxt.isNull() && !msgid_plural.isNull()) {
+                testMsgstr = xcatalog.translate(msgctxt, msgid, msgid_plural, n);
+            }
+        }
         if (!testMsgstr.isEmpty()) {
             // Translation found.
             language = testLanguage;
